bottomnavigation/bottomnavigation/src/main/java/com/microsoft/device/dualscreen/bottomnavigation/BottomNavigationView.kt

```diff
@@ -16,76 +17,132 @@ import android.view.MotionEvent
     private var hingeWidth = -1
 
     private var screenMode = ScreenMode.DUAL_SCREEN
-    private var currentScreenInfo: ScreenInfo? = null
+    private var foldingFeature: FoldingFeature? = null
 
     private var initialBackground: Drawable? = null
     private var startBtnCount: Int = -1
     private var endBtnCount: Int = -1
     private var defaultChildWidth = -1
 
-    private var onSwipeListener: OnSwipeListener = object : OnSwipeListener(context) {
-        override fun onSwipeLeft() {
-            super.onSwipeLeft()
-            if (allowFlingGesture) {
-                displayPosition = DisplayPosition.START
-            }
+    private fun initFoldingObs() {
+        val executor = ContextCompat.getMainExecutor(context)
+        val windowInfoRepository: WindowInfoRepository =
+            (context as Activity).windowInfoRepository()
+        job?.cancel()
+        job = CoroutineScope(executor.asCoroutineDispatcher()).launch {
+            windowInfoRepository.windowLayoutInfo
+                .mapNotNull { info -> getFoldingFeature(info) }
+                .distinctUntilChanged()
+                .collect { nextFeature ->
+                    onFoldingFeatureChange(nextFeature)
+                }
         }
+    }
 
-        override fun onSwipeRight() {
-            super.onSwipeRight()
-            if (allowFlingGesture) {
-                displayPosition = DisplayPosition.END
-            }
-        }
+    private fun getFoldingFeature(windowLayoutInfo: WindowLayoutInfo): FoldingFeature? {
+        return windowLayoutInfo.displayFeatures
+            .firstOrNull { feature -> feature is FoldingFeature } as? FoldingFeature
     }
 
-    private val screenInfoListener = object : ScreenInfoListener {
-        override fun onScreenInfoChanged(screenInfo: ScreenInfo) {
-            currentScreenInfo = screenInfo
-            setScreenParameters(screenInfo)
-            tryUpdateBackground()
-        }
+    open fun onFoldingFeatureChange(foldingFeature: FoldingFeature) {
+        this.foldingFeature = foldingFeature
+        setScreenParameters(foldingFeature)
+        tryUpdateBackground()
+
+        val changeBounds: Transition = ChangeBounds()
+        changeBounds.duration = 300L
+        changeBounds.interpolator = PathInterpolatorCompat.create(0.2f, 0f, 0f, 1f)
+        TransitionManager.beginDelayedTransition(this@BottomNavigationView, changeBounds)
+        requestLayout()
     }
 
+    private var onSwipeListener: OnSwipeListener =
+        object : OnSwipeListener(context) {
+            override fun onSwipeLeft() {
+                super.onSwipeLeft()
+                if (allowFlingGesture) {
+                    displayPosition =
+                        DisplayPosition.START
+                }
+            }
+
+            override fun onSwipeRight() {
+                super.onSwipeRight()
+                if (allowFlingGesture) {
+                    displayPosition = DisplayPosition.END
```

Comment:
nit. The style in this line is different than in previous if-else and they do the same (line 123-124)